daphne/src/testing.rs

```diff
@@ -374,28 +374,32 @@ impl DapHelper<BearerToken> for MockAggregator {
         let report_store = report_store_mutex_guard.deref_mut();
 
         for report_share in report_shares.iter() {
-            let bucket_info = BucketInfo::new(
-                self.tasks
-                    .get(task_id)
-                    .ok_or_else(|| DapError::fatal("task not found"))?,
-                task_id,
-                &report_share.nonce,
-            );
-
-            if !report_store.contains_key(&bucket_info) {
-                report_store.insert(bucket_info.clone(), ReportStore::new());
+            // Check whether the report is collected via the AggStore.
+            let bucket_info = BucketInfo::new(task_config, task_id, &report_share.nonce);
+            let collected = match agg_store.get(&bucket_info) {
+                Some(agg_store_state) => agg_store_state.collected,
+                None => false,
+            };
+            if collected {
```

Comment:
More idiomatic
```suggestion
            if matches!(agg_store.get(&bucket_info), Some(agg_store_state) if agg_store_state.collected) {
```

Reply:
Thanks, I didn't know about this way of writing.